src/Pact/Gas.hs

```diff
@@ -15,85 +15,95 @@ module Pact.Gas
  , putGas)
  where
 
+import Control.Monad(when)
 import Data.Text
 import Data.IORef
+import Pact.Types.Gas
 import Pact.Types.Runtime
 import Control.Lens
 import Control.Arrow
 import Data.Word
 import Pact.Types.Pretty
 
 -- | Compute gas for some application or evaluation.
-computeGas :: Either (Info,Text) FunApp -> GasArgs -> Eval e Gas
+computeGas :: Either (Info,Text) FunApp -> GasArgs -> Eval e ()
 computeGas i args = do
   GasEnv {..} <- view eeGasEnv
   g0 <- getGas
   let
     (info,name) = either id (_faInfo &&& _faName) i
     g1 = runGasModel _geGasModel name args
-  let gUsed = g0 + g1
-  evalLogGas %= fmap ((renderCompactText' (pretty name <> ":" <> pretty args <> ":currTotalGas=" <> pretty gUsed),g1):)
-  putGas gUsed
-  if gUsed > fromIntegral _geGasLimit then
-    throwErr GasError info $ "Gas limit (" <> pretty _geGasLimit <> ") exceeded: " <> pretty gUsed
-    else return gUsed
+  let gMillisUsed = g0 <> g1
+      gUsed = milliGasToGas gMillisUsed
+  evalLogGas %= fmap ((renderCompactText' (pretty name <> ":" <> pretty args <> ":currTotalGas=" <> pretty gUsed),milliGasToGas g1):)
+  putGas gMillisUsed
+  let (MilliGasLimit gasLimit) = _geGasLimit
```

Comment:
nit: I prefer calling the wrapped int `gasLimitMillis` when we destructure, so that the name carries the info about which units we're in. Mill less confusing.

Reply:
`(MilliGasLimit gasLimitMillis)` seems redundant, the pattern match here makes it clear what units its in, and even then, the type system would prevent us from doing something like comparing `Gas` to `MilliGas`.